js/components/generate/generate.js

```diff
@@ -79,58 +79,81 @@ class GenerateForm extends React.Component {
     fetch("/graph-generate", putData)
       .then(res => res.json())
       .then(data => {
-        const regionsList = []
-        const nodesList = []
-        const hybridsList = []
-        const boolsList = []
-        const edgesList = []
+        const labelsJSON = {}
+        const regionsJSON = {}
+        const nodesJSON = {}
+        const hybridsJSON = {}
+        const boolsJSON = {}
+        const edgesJSON = {}
+        const boolsStatus = {}
+        const nodesStatus = {}
         const parentsObj = {}
         const inEdgesObj = {}
         const childrenObj = {}
         const outEdgesObj = {}
         const storedNodes = new Set()
 
-        const labelsList = data.texts.filter(function (entry) {
-          // filter for mark percentages, allow preceding characters for potential geq
-          return entry.text.match(/.*[0-9]*%/g)
+        data.texts.forEach(entry => {
+          if (entry.text.match(/.*[0-9]*%/g)) {
+            labelsJSON[entry.rId] = entry
+            // filter for mark percentages, allow preceding characters for potential geq
```

Comment:
This is good, but the conventional placement would be to put it above the if condition